imports/plugins/core/core/server/no-meteor/resolvers/Query/primaryShop.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * @name "Query.primaryShop"
+ * @method
+ * @memberof Shop/GraphQL
+ * @summary Gets the primary shop
+ * @param {Object} parentObject - unused
+ * @param {Object} args - unused
+ * @param {Object} context - an object containing the per-request state
+ * @return {Promise<o>} The shop, based on the domain in ROOT_URL
+ */
+export default async function primaryShopId(_, __, context) {
+  const shopId = await context.queries.primaryShopId(context.collections);
+  const shop = await context.queries.shopById(context, shopId);
+  return shop;
```

Comment:
This is nice and simple, but it does two `Shops.findOne` calls when we often could do just one. You can make a separate `primaryShop` query function that does:

```js
const domain = url.parse(context.rootUrl).hostname;
let shop = await Shops.findOne({ domains: domain });
if (!shop) {
  shop = await Shops.findOne({ shopType: "primary" });
}
return shop;
```

And then call that in this resolver